### PR TITLE
[buffermgr] Support maximum port headroom checking

### PR DESCRIPTION
**Pull request by stephenxs**
**What I did**
Check the accumulative port headroom against the port's maximum headroom

On Mellanox platform, this PR depends on PR [#6566](https://github.com/Azure/sonic-buildimage/pull/6566) to be merged. In that PR the required SAI attribute is supported.
On other platforms, there is no dependency.

Signed-off-by: Stephen Sun <stephens@nvidia.com>

**Why I did it**
On some platforms, the SAI will notify orchagent shut down if a headroom size that causes the accumulative port headroom to exceed its limit is programmed to SAI.
To avoid that, we need to check this before programming it to SAI.

**How I verified it**
Run the regression test.

**Details if related**
- Fetch the maximum port headroom via fetching the port attribute `SAI_PORT_ATTR_QOS_MAXIMUM_HEADROOM_SIZE` when the orchagent starts and push the data into STATE_DB
- Check the accumulative port headroom against the maximum headroom in buffer_check_headroom_<vendor>.lua

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012

### CODE REVIEW

**Comment by liat-grozovik** [comment]
@stephenxs I believe this need to be merged only after Mellanox SAI 1.18.1.0 is merged. If so, can you please connect it to the PR?

**Comment by liat-grozovik** [comment]
@neethajohn can you please review?

File: orchagent/portsorch.cpp [hunk 1]
```diff
@@ -3252,6 +3255,25 @@ void PortsOrch::initializePriorityGroups(Port &port)
...
+void PortsOrch::initializePortMaximumHeadroom(Port &port)
+{
+    sai_attribute_t attr;
+
+    attr.id = SAI_PORT_ATTR_QOS_MAXIMUM_HEADROOM_SIZE;
+
+    sai_status_t status = sai_port_api->get_port_attribute(port.m_port_id, 1, &attr);
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_NOTICE("Unable to get number of priority groups for port %s rv:%d, ignored", port.m_alias.c_str(), status);
```
Comment (neethajohn) [inline orchagent/portsorch.cpp:3267]: Fix the error msg to be inline with the operation performed

**Comment by stephenxs** [comment]
> @stephenxs I believe this need to be merged only after Mellanox SAI 1.18.1.0 is merged. If so, can you please connect it to the PR?

Done

File: orchagent/portsorch.cpp (hunk 1 above)
Comment (stephenxs) [reply orchagent/portsorch.cpp]: Fixed.

**Review by neethajohn — APPROVED** [review]

**Review by liat-grozovik — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@daall can you please add 202012 label?

**Pull request merged**